apps/total-typescript/src/lib/exercises.ts

```diff
@@ -3,13 +3,14 @@ import groq from 'groq'
 import z from 'zod'
 
 export const ExerciseSchema = z.object({
-  _id: z.string(),
+  _id: z.string().optional(),
+  _key: z.string().optional(),
```

Comment:
we've got two types of exercises and one has an _id and the other has a _key

technically both drive these from `SanityDocument` type and in our presentation the `Exercise` instances with a `key` are actually the `/solution`

we can type this better I think.

Reply:
`z.object({ _id: z.string() }).or(z.object({ _key: z.string() }))`

?

Reply:
Riffing on @mattpocock's idea of using `or()` plus handling all the keys shared between the two:

```typescript
const everything = z.object({ _id: z.string(), key: z.string(), shared: z.string() })
const unionSchema = everything.omit({ _id: true }).or(everything.omit({ key: true }));

type Union = z.infer<typeof unionSchema>
/*
type Union = {
    key: string;
    shared: string;
} | {
    _id: string;
    shared: string;
}
*/
```